lib/api/controllers/securityController.js

```diff
@@ -753,3 +904,140 @@ function formatUserSearchResult(kuzzle, formatter, raw) {
+    }
+    if (!Array.isArray(request.input.body.content.profileIds)) {
+      throw new BadRequestError(`${getRequestRoute(request)} must specify the body attribute "content.profileIds" of type "array".`);
+    }
+  }
+
+  pojoUser = request.input.body.content;
+  pojoUser._id = request.input.resource._id || uuid.v4();
+
+  if (restricted) {
+    pojoUser.profileIds = kuzzle.config.security.restrictedProfileIds;
+  }
+
+  return Bluebird.coroutine(persistUserAndStrategies)(kuzzle, request, pojoUser);
+}
+
+/**
+ * @param {Kuzzle} kuzzle
+ * @param {Request} request
+ * @param {object} pojoUser
+ */
+function* persistUserAndStrategies (kuzzle, request, pojoUser) {
+  const
+    user = new User(),
+    loadedUser = yield Bluebird.resolve(kuzzle.repositories.user.load(pojoUser._id));
+  let
+    strategies = [],
+    errorStep = null,
+    creationError,
+    createdUser,
+    errors;
+
+  if (loadedUser !== null && loadedUser._id !== kuzzle.repositories.user.anonymous()._id) {
+    throw new BadRequestError(`${getRequestRoute(request)} user "${pojoUser._id}" already exist.`);
+  }
+
+  if (request.input.body.credentials && Object.keys(request.input.body.credentials).length > 0) {
+    strategies = Object.keys(request.input.body.credentials);
+
+    for (let i = 0; i < strategies.length; i++) {
+      const strategy = strategies[i];
+      let validateMethod;
+
+      if (kuzzle.pluginsManager.listStrategies().indexOf(strategies[i]) === -1) {
+        throw new BadRequestError(`${getRequestRoute(request)} strategy "${strategies[i]}" is not a known strategy.`);
+      }
+
+      validateMethod = kuzzle.pluginsManager.getStrategyMethod(strategy, 'validate');
+
+      try {
+        yield validateMethod(request, request.input.body.credentials[strategy], pojoUser._id, request.input.args.strategy, false);
+      }
+      catch (error) {
+        throw new BadRequestError(error);
+      }
+    }
+
+    for (let i = 0; i < strategies.length; i++) {
+      const
+        strategy = strategies[i],
+        createMethod = kuzzle.pluginsManager.getStrategyMethod(strategy, 'create');
+
+      try {
+        yield createMethod(request, request.input.body.credentials[strategy], pojoUser._id, request.input.args.strategy);
+      }
+      catch (error) {
+        errorStep = i;
+        creationError = error;
+        break;
+      }
+    }
+  }
+
+  if (errorStep === null) {
+    try {
+      createdUser = yield kuzzle.repositories.user.hydrate(user, pojoUser)
+        .then(modifiedUser => kuzzle.repositories.user.persist(modifiedUser, {database: {method: 'create'}}))
+        .then(modifiedUser => formatProcessing.formatUserForSerialization(kuzzle, modifiedUser));
+    }
+    catch (error) {
+      errorStep = strategies.length - 1;
+      creationError = error;
+    }
+  }
+
+  if (errorStep === null) {
+    return Bluebird.resolve(createdUser);
+  }
+
+  errors = [creationError];
+
+  // We try to delete the strategy that was in error as well
```

Comment:
Because the error could have happen during the return of the result of ES, and could have been processed on ES side.